app/scenes/select_server/select_server.js

```diff
@@ -21,22 +21,46 @@ import logo from 'assets/images/logo.png';
 import Client from 'service/client';
 import RequestStatus from 'service/constants/request_status';
 
-export default class SelectServer extends Component {
+import {isValidUrl, stripTrailingSlashes} from 'app/utils/url';
+
+export default class SelectServer extends PureComponent {
     static propTypes = {
-        serverUrl: React.PropTypes.string.isRequired,
-        server: React.PropTypes.object.isRequired,
-        actions: React.PropTypes.object.isRequired
+        serverUrl: PropTypes.string.isRequired,
+        server: PropTypes.object.isRequired,
+        actions: PropTypes.object.isRequired
     };
 
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            error: null
+        };
+    }
+
     onClick = () => {
-        Client.setUrl(this.props.serverUrl);
+        const url = this.props.serverUrl;
+        let error = null;
```

Comment:
Does ```error``` need to be declared at this point? Looks like it's only used in the ```else``` block. Could also move ```this.setState({...})``` to the else block as well since we only need to call it when an error is present.